x-pack/plugins/search_synonyms/public/plugin.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, type Subscription } from 'rxjs';
+
+import { AppStatus } from '@kbn/core/public';
+import type {
+  AppUpdater,
+  CoreSetup,
+  Plugin,
+  PluginInitializerContext,
+  AppMountParameters,
+  CoreStart,
+} from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { PLUGIN_ID, PLUGIN_NAME, PLUGIN_PATH } from '../common';
+import {
+  AppPluginSetupDependencies,
+  AppPluginStartDependencies,
+  SearchSynonymsConfigType,
+  SearchSynonymsPluginSetup,
+  SearchSynonymsPluginStart,
+} from './types';
+import { SYNONYMS_UI_FLAG } from '.';
+
+export class SearchSynonymsPlugin
+  implements Plugin<SearchSynonymsPluginSetup, SearchSynonymsPluginStart>
+{
+  private config: SearchSynonymsConfigType;
+  private readonly appUpdater$ = new BehaviorSubject<AppUpdater>(() => ({}));
+  private licenseSubscription?: Subscription;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.config = initializerContext.config.get<SearchSynonymsConfigType>();
+  }
+
+  public setup(
+    core: CoreSetup<AppPluginStartDependencies, SearchSynonymsPluginStart>,
+    _: AppPluginSetupDependencies
+  ): SearchSynonymsPluginSetup {
+    if (!this.config.ui?.enabled && !core.uiSettings.get<boolean>(SYNONYMS_UI_FLAG, false)) {
+      return {};
+    }
+    core.application.register({
+      id: PLUGIN_ID,
+      appRoute: '/app/elasticsearch/synonyms',
+      title: PLUGIN_NAME,
+      deepLinks: [
+        {
+          id: 'synonyms',
+          path: PLUGIN_PATH,
```

Comment:
`PLUGIN_PATH = '/synonyms'` so the full URL path is going to be `/app/elasticsearch/synonyms/synonyms` should that be `/app/elasticsearch/synonyms/list` or something else? the double `/synonyms` seems like it could be better.